src/engine.rs

```diff
@@ -700,14 +700,43 @@ mod tests {
             .prefix("test_empty_engine")
             .tempdir()
             .unwrap();
-        let mut sub_dir = PathBuf::from(dir.as_ref());
-        sub_dir.push("raft-engine");
-        let cfg = Config {
-            dir: sub_dir.to_str().unwrap().to_owned(),
-            ..Default::default()
-        };
-        RaftLogEngine::open_with_file_system(cfg, Arc::new(ObfuscatedFileSystem::default()))
-            .unwrap();
+        {
+            let mut sub_dir = PathBuf::from(dir.as_ref());
+            sub_dir.push("raft-engine");
+            let cfg = Config {
+                dir: sub_dir.to_str().unwrap().to_owned(),
+                ..Default::default()
+            };
+            RaftLogEngine::open_with_file_system(cfg, Arc::new(ObfuscatedFileSystem::default()))
+                .unwrap();
+        }
+        {
+            // Start engine with prefilled recycling. Testing too many
+            // prefilled files
+            let cfg = Config {
+                dir: dir.path().to_str().unwrap().to_owned(),
+                target_file_size: ReadableSize(1),
+                enable_log_recycle: true,
+                prefill_for_recycle: true,
+                ..Default::default()
+            };
+            if let Some(max_fd_limit) = fdlimit::raise_fd_limit() {
+                if cfg.recycle_capacity() >= max_fd_limit as usize {
+                    // It should return Error **open too many files**.
```

Comment:
Again, where is the error raised? I didn't find any using the key word "too many files".

Reply:
Anyway, it's a bad idea to enforce a limit using `recycle_capacity`. (1) number of opened files will not be affected by whether or not `recycle_capacity` is capped (2) even if append queue files are capped, rewrite queue files will grow out of limit too.

Reply:
> Again, where is the error raised? I didn't find any using the key word "too many files".

When prefilling with recycled files by setting `prefill_recycle_logs`, it will reach this error by setting `target_file_size == 1`.

Reply:
> Anyway, it's a bad idea to enforce a limit using `recycle_capacity`. (1) number of opened files will not be affected by whether or not `recycle_capacity` is capped (2) even if append queue files are capped, rewrite queue files will grow out of limit too.

Seems that this case is redundant and unnecessary. Originally, i just wanna add a case to mock the scenario where users set a huge `recycle_capacity` and `Engine` fails to be open as it encounters `too many open files` error.